sesman/session.c

```diff
@@ -676,6 +678,19 @@ session_start_fork(tbus data, tui8 type, struct SCP_SESSION *s)
                 g_snprintf(text, 255, "%d", g_cfg->sess.kill_disconnected);
                 g_setenv("XRDP_SESMAN_KILL_DISCONNECTED", text, 1);
 
+                /* prepare the Xauthority stuff */
+                if (g_getenv("XAUTHORITY") != NULL)
+                {
+                    g_snprintf(authfile, 255, "%s", g_getenv("XAUTHORITY"));
+                }
+                else
+                {
+                    g_snprintf(authfile, 255, "%s", ".Xauthority");
+                }
+
+                /* Add the entry in XAUTHORITY file */
+                add_xauth_cookie(display, authfile);
```

Comment:
Running with impaired security is a bad choice. Let's exit here on error with `g_exit(1);`

I would rather have users fix their xauth than have them run xrdp insecurely without knowing it.